virtual-desktop/src/app/i18n/translation-loader.service.ts

```diff
@@ -0,0 +1,94 @@
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+import {
+  Injectable,
+  LOCALE_ID,
+  StaticProvider,
+  TRANSLATIONS,
+  TRANSLATIONS_FORMAT,
+} from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/catch';
+import { LanguageLocaleService } from './language-locale.service';
+import * as Rx from 'rxjs/Rx';
+
+@Injectable()
+export class TranslationLoaderService {
+  constructor(
+    private languageLocaleService: LanguageLocaleService
+  ) {
+  }
+
+  getTranslationProviders(plugin: ZLUX.Plugin): Promise<StaticProvider[]> {
+    // Get the language id from the global
+    // According to Mozilla.org this will work well enough for the
+    // browsers we support (Chrome, Firefox, Edge, Safari)
+    // https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/language
+    // NOTES:
+    // 1. The desktop can override the browser language and locale preferences (see https://github.com/zowe/zlux-app-manager/issues/10),
+    //    so we don't just use the navigator language, but go through a "globalization" interface here.
+    // 2. Per the design of the above implementation (https://github.com/zowe/zlux-app-manager/pull/21), "true locale" can be separate
+    //    from language.
+    //    That pull request takes into account the subtleties about "locale" as discussed here:
+    //    https://www.w3.org/International/questions/qa-accept-lang-locales
+    //
+    // SUMMARY: The one part ('en') or two part ('en-US') language is treated *here* as *just language*, Maybe a specific sub language
+    //          e.g., en-GB has different spellings, so the "language" aspect of the second part can be important in choosing a template,
+    //          separate from implications for currency and decimal separator.
+    // MERGE QUESTION: should be put this in polyfills? abstract it somewhere? etc.?
+    const language: string = this.languageLocaleService.getLanguage();
+    // return no providers if fail to get translation file for language
+    const noProviders: StaticProvider[] = [];
+    // No language or U.S. English: no translation providers
+    if (this.languageLocaleService.isConfiguredForDefaultLanguage()) {
+      return Promise.resolve(noProviders);
+    }
+    const baseLanguage = this.languageLocaleService.getBaseLanguage();
+    // ex.: messages.es-ES.xlf
+    const translationFileURL = this.getTranslationFileURL(plugin, language);
+    // ex.: messages.es.xlf
+    const fallbackTranslationFileURL = baseLanguage !== language ? this.getTranslationFileURL(plugin, baseLanguage) : null;
+    return this.loadTranslations(translationFileURL)
+      .catch(err => (fallbackTranslationFileURL != null) ? this.loadTranslations(fallbackTranslationFileURL) : Observable.throw(err))
```

Comment:
Please issue a WARNING log message when "falling back":
`Failed to load language file ${ translationFileURL }, using ${fallbackTranslationFileURL }. Error ${err.something()}`

Reply:
I added such warnings. Please take a look at https://github.com/lchudinov/zowe-zlux-app-manager/commit/6251903aef6e329066fc240206125ad258777474.

Reply:
I added such warnings. Please take a look at https://github.com/lchudinov/zowe-zlux-app-manager/commit/6251903aef6e329066fc240206125ad258777474.

Reply:
I added such warnings. Please take a look at https://github.com/lchudinov/zowe-zlux-app-manager/commit/6251903aef6e329066fc240206125ad258777474.

Reply:
I added such warnings. Please take a look at https://github.com/lchudinov/zowe-zlux-app-manager/commit/6251903aef6e329066fc240206125ad258777474.

Reply:
I added such warnings. Please take a look at https://github.com/lchudinov/zowe-zlux-app-manager/commit/6251903aef6e329066fc240206125ad258777474.

Reply:
I added such warnings. Please take a look at https://github.com/lchudinov/zowe-zlux-app-manager/commit/6251903aef6e329066fc240206125ad258777474.